kerl

```diff
@@ -754,7 +668,128 @@ _flags() {
+# is an array with packages, and then create a unique probe
+# command, in variable _KPP_<distro>_probe to check from the package manager.
+# It should return 0 if package installation is Ok, non-0 otherwise.
+
+_KPP_alpine_pkgs="openssl-dev make automake autoconf ncurses-dev gcc g++"
+_KPP_alpine_probe="_apk"
+
+_KPP_debian_pkgs="libssl-dev make automake autoconf libncurses5-dev gcc g++"
+_KPP_debian_probe="_dpkg"
+
+_KPP_fedora_pkgs="openssl-devel make automake autoconf ncurses-devel gcc g++"
+_KPP_fedora_probe="_rpm"
+
+_KPP_linuxmint_pkgs="libssl-dev make automake autoconf libncurses5-dev gcc g++"
+_KPP_linuxmint_probe="_dpkg"
+
+_KPP_rhel_pkgs="openssl-devel make automake autoconf ncurses-devel gcc gcc-c++"
+_KPP_rhel_probe="_rpm"
+
+_KPP_ubuntu_pkgs="libssl-dev make automake autoconf libncurses5-dev gcc g++"
+_KPP_ubuntu_probe="_dpkg"
+
+parse_os_release() {
+    # $1: path to os-release
+    # returns:
+    # - 2 if release ID is not found in os-release file
+    os_release_id0=$(\grep "^ID=" "$1")
+    os_release_id=$(echo "${os_release_id0}" | \sed 's|.*=||' | \sed 's|"||g')
+    os_release_pretty_name0=$(\grep "^PRETTY_NAME=" "$1")
+    os_release_pretty_name=$(echo "${os_release_pretty_name0}" | \sed 's|.*=||' | \sed 's|"||g')
+    echo "[packages] Found ${os_release_pretty_name} with ID ${os_release_id}" >>"$LOGFILE"
+    if [ -z "${os_release_id}" ]; then
+        echo "[packages] Release ID not found in $1" >>"$LOGFILE"
+        echo "2"
+    else
+        echo "[packages] Release ID found in $1: ${os_release_id}" >>"$LOGFILE"
+        echo "${os_release_id}"
+    fi
+}
+
+get_id_from_os_release_files() {
+    # returns:
+    # 1 - if no release files exist
+    files="/etc/os-release /usr/lib/os-release"
+    for file in ${files}; do
+        if [ -f "${file}" ]; then
+            parsed=$(parse_os_release "${file}")
+            if [ "${parsed}" != "2" ]; then
+                break
+            fi
+        fi
+    done
+
+    if [ -z "${parsed}" ]; then
+        echo "[packages] Found no file in: ${files}. Bailing out..." >>"$LOGFILE"
+        echo "1"
+    fi
+
+    echo "${parsed}"
+}
+
+probe_pkgs() {
+    os_release_id=$(get_id_from_os_release_files)
+
+    if [ "${os_release_id}" = "1" ]; then
+        msg="[packages] Unable to determine Linux flavor (no release files); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    elif [ "${os_release_id}" = "2" ]; then
+        msg="[packages] Unable to determine Linux flavor (no ID); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    fi
+
+    kpp=$(eval echo \$_KPP_"${os_release_id}"_pkgs)
+    if [ -n "${kpp}" ]; then
+        echo "[packages] Found package declarations for ${os_release_id}. Linux flavor is known." >>"$LOGFILE"
+        echo "[packages] Acting on known Linux distro ${os_release_id}" >>"$LOGFILE"
```

Comment:
Seems like you could combine these messages.

`[packages] Your linux seems to be ${os_release_id}.` ???

Reply:
Yeah, I guess so. A previous implementation had them separated into different code paths, but one I joined it into this single `if` didn't take care of it. 👍

Reply:
7fa76c3, then f6deb2b to not have it called "flavor" sometimes and "distro" some other times 😄

Reply:
Yeah we should drop them or stick to one consistent word :)